drift/lib/src/runtime/query_builder/expressions/case_when.dart

```diff
@@ -1,30 +1,56 @@
-import 'package:meta/meta.dart';
-
 import '../query_builder.dart';
 
+/// A `CASE WHEN` expression (without base expression) in sqlite.
```

Comment:
The documentation here should link `[Expression.caseMatch]` to mention the variant that does have a base expression.

It should also mention what `when` does, e.g. "this expression evaluates to the first value in [when] for which its key evaluates to `true`".